### DISCUSSION TITLE
How to solve "System.EntryPointNotFoundException"?

### DISCUSSION THREAD

**Discussion by MaiastraMan** [microsoft/microsoft-ui-xaml#8497] [Q&A]
I have encountered the following error when I try to run my WinUI3 project.

`System.EntryPointNotFoundException: 'Unable to find an entry point named 'XamlCheckProcessRequirements' in DLL 'Microsoft.ui.xaml.dll'.'`

Specifically, the run options I am using are "Release" "X64" "MyAppName (Package)" "Local Machine". I am using Version 17.6.0 of Visual Studio, and this problem started when I changed my configuration from "Release" to "Debug" (I tried changing it back but this made no difference).

I have tried the following solutions:

1. All actions suggested in [this thread](https://stackoverflow.com/questions/72324559/winui3-entrypointnotfoundexception-after-vs-17-1-2-update). Specifically, these were deleting the following folder: C:\Users\UserName.nuget\packages\microsoft.ui.xaml And ensuring my .csproj (and wapproj) files contain the correct elements.
2. Ensured that Visual Studio and the WinAppSDK installations were the latest versions.
3. Started a new project, using the  "Blank App, Packaged with Windows Application Packaging Project (WinUI3 in Desktop)" template. This worked when I ran the blank app. Then, I used NuGet package manager to add the Microsoft.UI.Xaml version 2.8.4. After I added this, the error reoccurred.
4. Started procmon, then ran my package. This showed that the dll was successfully loaded from the following folder: `C:\Program Files\WindowsApps\Microsoft.UI.Xaml.2.8_8.2305.5001.0_x64__8wekyb3d8bbwe\Microsoft.UI.Xaml` I inspected the .dll file in this folder, and found that it was 2.8.2305.5001. This conflicted with my NuGet Package manager version, which was 2.8.4. Consequently I tried "upgrading" my NuGet Package version to 2.8.2, however the error persisted.

If anyone has any thoughts on how to proceed from here, I would greatly appreciate if you would let me know.

**Comment by DarranRowe** [comment]
To put it simply, WinUI 2 and WinUI 3 cannot coexist in the same process. Since there are now two Microsoft.UI.Xaml.dll libraries referenced, it is completely up to luck what version it will bind to with the check for this function. It is obviously loading the WinUI 2 version, and this doesn't export this function.

**Reply by MaiastraMan** [reply]
Thank you very much for your reply, to let you know, I actually managed to solve this. It turned out that for some unknown reason, Visual Studio started adding the following line to my `.csproj` file: `<PackageReference Include="Microsoft.UI.Xaml" Version="2.8.4" />`. I removed this line and it started working again. Just wanted to  mention it here in case it helps anyone else.

**Reply by HerraHak** [reply]
I've got to thank you here.  I've been having this problem for three days now and I couldn't figure out why it didn't work.  I uninstall VS 2022, uninstall .net, cleared my computer from any trace of previous installations and yet it failed.
Then I saw your comment, deleted the line and bam it works.

Thank you, Thank you !!!!!
